### PR TITLE
Setup docker

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Add `setup_docker` role for integration tests.

##### ISSUE TYPE

- Test Pull Request

##### COMPONENT NAME
mssql_script
keycloak_group

### CODE REVIEW

File: tests/integration/targets/mssql_script/tasks/main.yml [hunk 1]
```diff
@@ -13,13 +13,33 @@
...
+- name: Start container
+  community.docker.docker_container:
+    name: mssql-test
+    image: "mcr.microsoft.com/mssql/server:2019-latest"
+    env:
+      ACCEPT_EULA: "Y"
+      SA_PASSWORD: "{{ mssql_login_password }}"
+      MSSQL_PID: Developer
+    ports:
+      - "{{ mssql_port }}:1433"  # {{ mssql_port }}"
```
Comment (russoz) [inline tests/integration/targets/mssql_script/tasks/main.yml:31]:
```suggestion
      - "{{ mssql_port }}:1433"
```

File: tests/integration/targets/mssql_script/tasks/main.yml [hunk 2]
```diff
@@ -13,13 +13,33 @@
...
+
 - name: Check default ports
-  wait_for:
+  ansible.builtin.wait_for:
     host: "{{ mssql_host }}"
     port: "{{ mssql_port }}"
     state: started         # Port should be open
-    delay: 0               # No wait before first check (sec)
-    timeout: 3             # Stop checking after timeout (sec)
+    delay: 10              # No wait before first check (sec)
```
Comment (russoz) [inline tests/integration/targets/mssql_script/tasks/main.yml:41]:
```suggestion
    delay: 10              # Wait 10 secs before first check
```

File: tests/integration/targets/setup_docker/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+The most common use case is to use `community.docker.docker_container` to start a container, as in the example above. It is likely that `community.docker.docker_compose` can be used as well, although this has **not been tested** yet.
+
+
+Recommendations
+---------------
+
+* Don't forget to publish the service ports when starting the container
+* Take into consideration that the services inside the container will take a while to get started. Use both/either `ansible.builtin.wait_for` to check for the availability of the network port and/or `retries` on the first task effectively using those services
+* As a precautionary measure, start using the role in a test that is marked either `disabled` or `unsupported`, and move forward from there. If intent on running a test that uses docker in the CI execution, please make sure that intention is mentioned with some anticipation to the collection maintainers as it can significantly increase the load and time length of the tests.
```
Comment (russoz) [inline tests/integration/targets/setup_docker/README.md:65]:
```suggestion
* As a precautionary measure, start using the role in a test that is marked either `disabled` or `unsupported`, and move forward from there.
```

**Comment by russoz** [comment]
the error in the CI seems to be unrelated to this

File: tests/integration/targets/mssql_script/tasks/main.yml [hunk 4]
```diff
@@ -1,9 +1,4 @@
 ---
-####################################################################
-# WARNING: These are designed specifically for Ansible tests       #
-# and should not be used as examples of how to write Ansible roles #
-####################################################################
-
```
Comment (felixfontein) [inline tests/integration/targets/mssql_script/tasks/main.yml:6]: Why remove the above?

Comment (russoz) [reply tests/integration/targets/mssql_script/tasks/main.yml:6]: I assumed it was only boilerplate from the template, but happy to roll those back in

Comment (felixfontein) [reply tests/integration/targets/mssql_script/tasks/main.yml:6]: It's a generic boilerplate we add to all integration tests.

**Comment by felixfontein** [comment]
Restart CI.

File: tests/integration/targets/mssql_script/tasks/main.yml (hunk 4 above)
Comment (russoz) [reply tests/integration/targets/mssql_script/tasks/main.yml]: Rolled back in

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for working on this!

**Pull request merged**